docs/modules/ROOT/pages/explanations/dev/iamkey-reconciliation.adoc

```diff
@@ -0,0 +1,28 @@
+= Development: IAMKey Reconciliation
+
+This page gives a brief overview how `IAMKeys` are managed in the provider.
+
+Note that these diagrams are on a high-level and don't visualize the path for every possible situation.
+
+== Creating IAMKeys
+
+image::iamkey-create.drawio.svg[]
+
+- In exoscale.com API, display names are not unique, there can be multiple IAM keys with different IAM key IDs that share the same display name.
+- During the first reconciliation we do not check if an iam key with the desired display name exists, since we don't know the iam key ID.
```

Comment:
```suggestion
- During the first reconciliation we do not check if an IAM key with the desired display name exists, since we don't know the iam key ID.
```